### DISCUSSION TITLE
[Idea] Switch getSpecialProducts() function to protected

### DISCUSSION THREAD

**Discussion by komunikelliot** [PrestaShop/PrestaShop#35247] [Ideas]
### The problem

Actually, the function is set as a private function, which restrict overriding the function from the prestashop overrides system.
module ps_newproducts use a similar function and the function is set as protected, which allow overriding the function.
This is usefull when you want to customize the way that the module get the products or if you want to manipulate the results before returning them.

### The solution

Change
  private function getSpecialProducts()
to
  protected function getSpecialProducts()

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
Hello @komunikelliot 

It would be ok, if you are motivated to create a pull request, feel free to do that 👍🏻
